R/colour-manip.R

```diff
@@ -175,29 +186,71 @@ NULL
 #' @export
 #' @rdname colour_manip
 col_shift <- function(col, amount = 10) {
+  UseMethod("col_shift")
+}
+
+#' @export
+col_shift.default <- function(col, amount = 10) {
   input  <- recycle_common(col = col, amount = amount)
   new <- farver::decode_colour(input$col, alpha = TRUE, to = "hcl")
   new[, "h"] <- (new[, "h"] + input$amount) %% 360
   farver::encode_colour(new, new[, "alpha"], from = "hcl")
 }
 
+#' @export
+col_shift.scales_pal <- function(col, amount = 10) {
+  wrap_col_adjustment(col, col_shift, list(amount = amount))
+}
+
 #' @export
 #' @rdname colour_manip
 col_lighter <- function(col, amount = 10) {
+  UseMethod("col_lighter")
+}
+
+#' @export
+col_lighter.default <- function(col, amount = 10) {
   input <- recycle_common(col = col, amount = amount)
   farver::add_to_channel(input$col, "l", input$amount, space = "hsl")
 }
 
+#' @export
+col_lighter.scales_pal <- function(col, amount = 10) {
+  wrap_col_adjustment(col, col_lighter, list(amount = amount))
+}
+
 #' @export
 #' @rdname colour_manip
 col_darker <- function(col, amount = 10) {
-  input <- recycle_common(col = col, amount = amount)
-  farver::add_to_channel(input$col, "l", -input$amount, space = "hsl")
+  col_lighter(col, amount = -amount)
 }
 
 #' @export
 #' @rdname colour_manip
 col_saturate <- function(col, amount = 10) {
+  UseMethod("col_saturate")
+}
+
+#' @export
+col_saturate.default <- function(col, amount = 10) {
   input <- recycle_common(col = col, amount = amount)
   farver::add_to_channel(input$col, "s", input$amount, space = "hsl")
 }
+
+#' @export
+col_saturate.scales_pal <- function(col, amount = 10) {
+  wrap_col_adjustment(col, col_saturate, list(amount = amount))
+}
+
+wrap_col_adjustment <- function(inner, outer, args, call = caller_env()) {
+  if (!is_colour_pal(inner)) {
```

Comment:
Update this to use the new type checkers once it is merged in

Reply:
👍